src/mlpack/core/optimizers/adam/adam_impl.hpp

```diff
@@ -67,6 +70,9 @@ double Adam<DecomposableFunctionType>::Optimize(arma::mat& iterate)
   // Exponential moving average of gradient values.
   arma::mat m = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
 
+  // Initialize the exponentially weighted infinity norm for AdaMax optimizer.
+  arma::mat u = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
```

Comment:
I would do it right at line 74, 77 that way we don't have to check if it's already initialized in every iteration:

```
arma::mat u;
if (adamax)
{
  u = ...
}
```